tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -398,15 +389,90 @@ internal static bool TryParseTraceState(string header, out W3CTraceState traceSt
             }
         }
 
+        internal static void SplitTraceStateValues(string header, out string ddValues, out string additionalValues)
+        {
+            // header format: "[*,]dd=s:1;o:rum;t.dm:-4;t.usr.id:12345[,*]"
+            ddValues = string.Empty;
+            additionalValues = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(header))
+            {
+                return;
+            }
+
+            header = header.Trim();
+            int ddStartIndex;
+
+            if (header.StartsWith("dd=", StringComparison.Ordinal))
+            {
+                ddStartIndex = 0;
+            }
+            else
+            {
+                // if "dd=" is not at start of header, make sure we find the one preceded by comma
+                // in case there is something like "key1=valuedd=whatisthis,dd=..."
+                //                                                          ^ take this one
+                //                                            ^ ignore this one
+                ddStartIndex = header.IndexOf(",dd=", StringComparison.Ordinal);
+
+                if (ddStartIndex >= 0)
+                {
+                    // skip the ',' (but don't add 1 if ddStartIndex == -1)
+                    ddStartIndex++;
+                }
+            }
+
+            if (ddStartIndex < 0)
+            {
+                // "dd=" was not found in header
+                additionalValues = header;
+                return;
+            }
+
+            var ddEndIndex = header.IndexOf(',', ddStartIndex + 3);
+
+            if (ddEndIndex < 0)
+            {
+                // comma not found, "dd=" reaches the end of header
+                ddEndIndex = header.Length;
+            }
+
+            ddValues = header.Substring(ddStartIndex, ddEndIndex - ddStartIndex);
+
+            if (ddStartIndex == 0 && ddEndIndex == header.Length)
+            {
+                // "dd" was the only key, no additional values
+                additionalValues = string.Empty;
+            }
+            else if (ddStartIndex == 0)
+            {
+                // "dd" first, additional values later
+                additionalValues = header.Substring(ddEndIndex + 1, header.Length - ddEndIndex - 1);
+            }
+            else if (ddEndIndex == header.Length)
+            {
+                // additional values first, "dd" later
+                additionalValues = header.Substring(0, ddStartIndex - 1);
+            }
+            else
+            {
+                // additional values on both sides, "dd" in the middle
+                var otherValuesLeft = header.Substring(0, ddStartIndex - 1);
+                var otherValuesRight = header.Substring(ddEndIndex + 1, header.Length - ddEndIndex - 1);
+                additionalValues = string.Join(",", otherValuesLeft, otherValuesRight);
+            }
+
+            additionalValues = additionalValues.Trim();
```

Comment:
nitpicking, do we really care of trimming?

Reply:
I removed trimming of non-dd values so we pass them through as-is (except if the entire `tracestate` header is whitespace, then we ignore it).